pipeline/init.bats

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bats
+
+set -ueo pipefail
+
+setup() {
+    load "bats-helpers/bats-support/load"
+    load "bats-helpers/bats-assert/load"
+
+    export PATH="${BATS_TEST_DIRNAME}/../bin:${PATH}"
+
+    export CK8S_CONFIG_PATH="${BATS_TMPDIR}/ck8s-apps-config"
+    export CK8S_ENVIRONMENT_NAME=test
+    export CK8S_PGP_FP=529D964DE0BBD900C4A395DA09986C297F8B7757
+    export CK8S_CLOUD_PROVIDER=aws
+}
+
+teardown() {
+    rm -rf "${CK8S_CONFIG_PATH}"
+}
+
+@test "ck8s init requires CK8S_CONFIG_PATH" {
+    CK8S_CONFIG_PATH="" run ck8s init
+    assert_failure
+    assert_output --partial "Missing CK8S_CONFIG_PATH"
+}
+
+@test "ck8s init requires CK8S_ENVIRONMENT_NAME" {
+    CK8S_ENVIRONMENT_NAME="" run ck8s init
+    assert_failure
+    assert_output --partial "Missing CK8S_ENVIRONMENT_NAME"
+}
+
+@test "ck8s init requires valid CK8S_PGP_FP or valid CK8S_PGP_UID" {
+    unset CK8S_PGP_FP
+
+    run ck8s init
+    assert_failure
+    assert_output --partial "CK8S_PGP_FP and CK8S_PGP_UID can't both be unset"
+
+    CK8S_PGP_FP="123" run ck8s init
+    assert_failure
+    assert_output --partial "Fingerprint does not exist in gpg keyring"
+
+    CK8S_PGP_UID="asd" run ck8s init
+    assert_failure
+    assert_output --partial "Unable to get fingerprint from gpg keyring using UID"
+}
+
+@test "ck8s init requires CK8S_CLOUD_PROVIDER" {
+    CK8S_CLOUD_PROVIDER="" run ck8s init
+    assert_failure
+    assert_output --partial "Missing CK8S_CLOUD_PROVIDER"
+}
+
+@test "ck8s init checks supported cloud providers" {
+    CK8S_CLOUD_PROVIDER=foo run ck8s init
+    assert_failure
+    assert_output --partial "Unsupported cloud provider: foo"
+}
+
+@test "ck8s init checks supported flavors" {
+    CK8S_CLOUD_PROVIDER=aws CK8S_FLAVOR=foo run ck8s init
+    assert_failure
+    assert_output --partial "Unsupported flavor: foo"
+}
+
+@test "ck8s init is idempotent" {
+    # TODO: Convert this to parametric test if it gets implemented:
+    # https://github.com/bats-core/bats-core/issues/241
+    [ -n "${CK8S_SKIP_LONG_RUNNING_TESTS:-}" ] && skip
+    for cloud_provider in aws citycloud exoscale safespring; do
```

Comment:
How should we make sure that we remembers to update these as well when adding new cloud providers? Maybe we can add a checkbox on PR templates for now just to have some reminder?

Reply:
Yea I've thought about this as well. At first I added it to the GitHub actions matrix, that made things more spread out and I kind of like that the tests can be run as-is instead of having to be run multiple times to cover all cases.

What I did in devbox was putting all the providers in a common file so that you only had to set them once. That is at least better. I can add this.

Reply:
What about adding commands that lists the available providers and flavors?

```
$ ck8s providers
aws citycloud exoscale safespring
$ ck8s flavors
dev prod
```

Then we have a single source of truth.

Reply:
Yea, I like that idea

Reply:
Check https://github.com/elastisys/compliantkubernetes-apps/pull/493

Reply:
This is based on the above PR, hopefully this should make it less error-prone when new providers are added. :)